elastic/beats
# Add SASL/SCRAM support to the Kafka test container

## What does this PR do?

Expands the configuration of the Kafka test container so that it adds a new external port 9093 that uses SASL/SCRAM authentication.

## Why is it important?

As part of making SASL/SCRAM on Kafka GA, we want to include it in the automated tests, so our Kafka test container needs to support it (and include the certificates necessary for the client to connect).

## Checklist


## How to test this PR locally

Create and start the container:

```
docker build -t kafkatest testing/environments/docker/kafka
docker run -p 127.0.0.1:9092:9092/tcp -p 127.0.0.1:9093:9093/tcp --rm --name kafkacontainer -d kafkatest
```

Point `kafka` at localhost (this is only needed to access the kafka broker from the host system, not another container, since on the Docker network the broker identifies itself by that name): edit `/etc/hosts` to add the line `127.0.0.1 kafka`.

The server can now be accessed by beats on the host system, e.g. with this output configuration:

```yml
output.kafka:
  hosts: ["127.0.0.1:9093"]
  topic: "hello"
  protocol: "https"

  ssl.certificate_authorities: ["testing/environments/docker/kafka/certs/ca-cert"]
  sasl.mechanism: "SCRAM-SHA-512"

  username: "beats"
  password: "KafkaTest"
```

(the path in `certificate_authorities` should be modified if necessary to point at `ca-cert` in the test container directory).
